enumerable.rb

```diff
@@ -0,0 +1,116 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    var = to_a
+    result = []
+    i = 0
+    while i < var.length
+      result << yield(var[i])
+      i += 1
+    end
+    result
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    var = to_a
+    result = i = 0
+    while i < var.length
+      result = yield(var[i], i)
+      i += 1
+    end
+    result
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each { |i| arr.push(i) if yield(i) }
+    arr
+  end
+
+  def my_all?
+    var = to_a
+    var.empty?
+    if block_given?
+      my_each { |i| return false unless yield(i) }
+      return true
+    end
+    var.include?(nil) || var.include?(false) ? false : true
+  end
+
+  def my_any?
+    var = to_a
+    if block_given?
+      var.my_each { |i| return true if yield(i) }
+    else
+      var.my_each { |i| return true if i.nil? == false and i != false }
+    end
+    false
+  end
+
+  def my_none?
+    var = to_a
+    if block_given?
+      var.my_each { |i| return false if yield(i) }
+    else
+      var.my_each { |i| return false if i.nil? == false and i != false }
+    end
+    true
+  end
+
+  def my_count(*arg)
+    sum = 0
+    if block_given?
+      my_each { |i| sum += 1 if yield(i) }
+      sum
+    elsif arg.empty?
+      length
+    else
+      my_each { |i| sum += 1 if arg[0] == i }
+      sum
+    end
+  end
```

Comment:
- [ ] should have working `my_count` **(covered by automated tests)** identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_count

- [ ] returns the number of items in enum through enumeration::range

```
ruby
(5..50).my_count == (5..50).count => 46
```
- Yours retrieves a NameError
![image](https://user-images.githubusercontent.com/52209067/94222097-b7838780-fea9-11ea-8745-9ae878f7d877.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/94222117-c23e1c80-fea9-11ea-98bd-b9714ac23ba1.png)